packages/miniflare/test/plugins/core/inspector-proxy/index.spec.ts

```diff
@@ -0,0 +1,440 @@
+					unsafeInspectorProxy: true,
+				},
+				{
+					name: "extra-worker-a",
+					script: nullScript,
+					unsafeInspectorProxy: true,
+				},
+				{
+					name: "extra-worker-b",
+					script: nullScript,
+					unsafeInspectorProxy: true,
+				},
+			],
+		});
+		t.teardown(() => mf.dispose());
+
+		await mf.ready;
+
+		const res = await fetch("http://localhost:9212/json");
+		const inspectors = (await res.json()) as Record<string, string>[];
+
+		t.is(inspectors.length, 3);
+
+		t.is(inspectors[0]["description"], "workers");
+		t.is(inspectors[0]["title"], "Cloudflare Worker");
+		t.is(inspectors[0]["webSocketDebuggerUrl"], "ws://localhost:9212/");
+		t.is(inspectors[1]["description"], "workers");
+		t.is(inspectors[1]["title"], "Cloudflare Worker: extra-worker-a");
+		t.is(
+			inspectors[1]["webSocketDebuggerUrl"],
+			"ws://localhost:9212/extra-worker-a"
+		);
+		t.is(inspectors[2]["description"], "workers");
+		t.is(inspectors[2]["title"], "Cloudflare Worker: extra-worker-b");
+		t.is(
+			inspectors[2]["webSocketDebuggerUrl"],
+			"ws://localhost:9212/extra-worker-b"
+		);
+	}
+);
+
+test.serial(
+	"InspectorProxy: /json should provide a list of a multiple worker inspector with some filtered out",
+	async (t) => {
+		const mf = new Miniflare({
+			inspectorPort: 9212,
+			unsafeInspectorProxy: true,
+			workers: [
+				{
+					script: nullScript,
+					unsafeInspectorProxy: true,
+				},
+				{
+					name: "extra-worker-a",
+					script: nullScript,
+				},
+				{
+					name: "extra-worker-b",
+					script: nullScript,
+					unsafeInspectorProxy: true,
+				},
+			],
+		});
+		t.teardown(() => mf.dispose());
+
+		await mf.ready;
+
+		const res = await fetch("http://localhost:9212/json");
+		const inspectors = (await res.json()) as Record<string, string>[];
+
+		t.is(inspectors.length, 2);
+
+		t.is(inspectors[0]["description"], "workers");
+		t.is(inspectors[0]["title"], "Cloudflare Worker");
+		t.is(inspectors[0]["webSocketDebuggerUrl"], "ws://localhost:9212/");
+		t.is(inspectors[1]["description"], "workers");
+		t.is(inspectors[1]["title"], "Cloudflare Worker: extra-worker-b");
+		t.is(
+			inspectors[1]["webSocketDebuggerUrl"],
+			"ws://localhost:9212/extra-worker-b"
+		);
+	}
+);
+
+test.serial(
+	"InspectorProxy: should allow debugging a single worker",
+	async (t) => {
+		const mf = new Miniflare({
+			inspectorPort: 9212,
+			unsafeInspectorProxy: true,
+			workers: [
+				{
+					script: `
+						export default {
+							fetch(request, env, ctx) {
+								debugger;
+								return new Response("body");
+							}
+						}
+					`,
+					modules: true,
+					unsafeInspectorProxy: true,
+				},
+			],
+		});
+		t.teardown(() => mf.dispose());
+
+		await mf.ready;
+
+		// when the inspector proxy is created there are some ws message exchanges between it
+		// and the runtime, these can interfere with testing and make tests flaky, so we just
+		// wait a bit here to avoid such problems
+		await setTimeout(30);
```

Comment:
30 seems a very small amount.

Reply:
yes but it seems to be sufficient, at least for me locally, if this turns out to be an issue in CI we can increase it but as long as it doesn't introduce flakyness the smaller the time the better, right?